pkg/agent/nodeportlocal/k8s/npl_controller.go

```diff
@@ -53,30 +53,32 @@ const (
 )
 
 type NPLController struct {
-	portTable   *portcache.PortTable
-	kubeClient  clientset.Interface
-	queue       workqueue.RateLimitingInterface
-	podInformer cache.SharedIndexInformer
-	podLister   corelisters.PodLister
-	svcInformer cache.SharedIndexInformer
-	podToIP     map[string]string
-	nodeName    string
-	podIPLock   sync.RWMutex
+	portTable        *portcache.PortTable
+	kubeClient       clientset.Interface
+	queue            workqueue.RateLimitingInterface
+	podInformer      cache.SharedIndexInformer
+	podLister        corelisters.PodLister
+	svcInformer      cache.SharedIndexInformer
+	podToIP          map[string]string
+	nodeName         string
+	podIPLock        sync.RWMutex
+	SyncRuleInterval time.Duration
```

Comment:
I am not sure this needs to be an exported field.
You do use it in the integration test that you have, but the integration test is also responsible for configuring that field so it already knows the value.